src/pip/_internal/vcs/git.py

```diff
@@ -155,6 +163,7 @@ def resolve_revision(cls, dest, url, rev_options):
                 rev,
             )
 
+        assert rev is not None
```

Comment:
I had said to add this after the `rev = rev_options.arg_rev` above:

> The way I would handle this is add assert rev is not None after the rev = rev_options.arg_rev line above (which is true for the Git class if you look at the implementation of the arg_rev property).

The reasons is that's the point in this method at which we know the assertion is True.